### DISCUSSION TITLE
Heterogenous nucleation

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#19410] [Q&A Modules: Phase field] [answered]
I have system of metal electrolyte interface with metal phase growing (electrodeposition). I want add nucleation using phase field module at the solid liquid interface only (heterogenous nucleation). Can i use discrete nucleation map at only the interface region. If yes how can i specify the region with conditional operators.

**Comment by laagesen** [comment, marked as answer]
You will want to make the nucleation probability a function of position through the `DiscreteNucleationInserter` rather than `DiscreteNucleationMap`. You can create a material property and supply the name of the material property to the `probability` parameter of the `DiscreteNucleationInserter`. You can make the material property using a `DerivativeParsedMaterial` and make it a function that is non-zero only in the interface. For example, if the order parameter is phi and ranges from 0 to 1 in the interface, you could make a function phi^2 * (1-phi)^2 times some constant to control the rate.

**Reply by laagesen** [reply]
You could also use a conditional as you asked by using an if statement in the DerivativeParsedMaterial, so that the nucleation probability is non-zero only if 0.25 < phi < 0.75 or some other values of your choosing. You will probably need to play with the values you choose a bit to decide exactly where to set them. There should be some examples of using if statements in the `DerivativeParsedMaterial` functions but feel free to reach out again if you don't find anything.

**Reply by vermaprk** [reply]
It worked Thanks for the help
